### PR TITLE
[Chore](function) remove some unused member function of IFunctionBase

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
## Proposed changes
remove some unused member function of IFunctionBase
## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by BiteTheDDDDt** [comment]
run buildall

**Review by yiguolei — APPROVED** [review]

**Review by xinyiZzz — APPROVED** [review]
LGTM

**Review by jacktengg — APPROVED** [review]

File: be/src/vec/functions/function_java_udf.h [hunk 1]
```diff
@@ -89,9 +89,12 @@ class JavaFunctionCall : public IFunctionBase {
     PreparedFunctionPtr prepare(FunctionContext* context, const Block& sample_block,
                                 const ColumnNumbers& arguments, size_t result) const override {
         return std::make_shared<JavaUdfPreparedFunction>(
-                std::bind<Status>(&JavaFunctionCall::execute_impl, this, std::placeholders::_1,
-                                  std::placeholders::_2, std::placeholders::_3,
-                                  std::placeholders::_4, std::placeholders::_5),
+                [this](auto&& PH1, auto&& PH2, auto&& PH3, auto&& PH4, auto&& PH5) {
```
Comment (zclllyybb) [inline be/src/vec/functions/function_java_udf.h:92]: For most situations, we can replace `auto&&` suggested by clang-tidy with concrete type. there's no need to use template here.

**Review by zclllyybb — APPROVED** [review]
LGTM. some minor case could be fixed in future

**Pull request merged**